app/assets/javascripts/plate-map/add-data-on-change.js

```diff
@@ -151,6 +178,18 @@ var plateLayOutWidget = plateLayOutWidget || {};
         var checkboxes = this.globalSelectedAttributes.slice(); 
         var selectedAreas = this.selectedAreas.slice(); 
         var focalWell = this.focalWell;
+        var selectedWells;
+
+        return {
+          "derivative": derivative,
+          "checkboxes": checkboxes,
+          "selectedAreas": selectedAreas,
+          "focalWell": focalWell
+        };
+      },
+
+      getCurrentPlate: function(){
```

Comment:
Why are there two functions that do nearly the same thing? Can't we combine them, or at least have getCurrentPlate build on createObject?

Reply:
The main reason here is that ebase is using the createObject() function, I refactored so that createObject() returns the current state and getCurrentPlate() is used internally in platemap